### DISCUSSION TITLE
How to write Logger output to a file?

### DISCUSSION THREAD

**Discussion by prgofficial** [adonisjs/core#4175] [Help]
I tried many configurations in config/app.ts. Still, I am not able to write the log to a file. 
I am using Adonijs version 5.9.0. Please help

**Comment by Julien-R44** [comment]
Please show what you've tried. Do you use Pm2? If yes then the logs are already sent to a file

otherwise, you could use the pino/file transport for this : https://getpino.io/#/docs/transports?id=pinofile

**Reply by prgofficial** [reply]
I'am using node ace to serve. in future, i will be using docker.
I tried setting some configuration in config/app.ts which i got from surfing the net.
like inside LoggerConfig;
  file: {
    enabled: true,
    filename: 'adonis.log',
    level: 'info',
  },

or 

  enabled: true,
  transport: 'file',
  level: 'info',
  file: {
    filename: 'adonis.log',
  },

  enabled: true,
  driver: 'file',
  level: 'info',
  file: {
    filename: 'adonis.log',
  },

etc. but none of these seems to be working. And how can use pino to write to file? It will be better to use the same Logger and set some configurations to write it to file. is that possible?

**Reply by Julien-R44** [reply]
Adonis uses pino as its logger. You can just run your adonis process and pipe the output in a file :

```
node build/server.js >> /mylogfile.txt
```

**Reply by prgofficial** [reply]
So what about in development? And also, what are the possibilities of date based logs? All logs in same file is not a good option. That's why I checked for some configuration inside adonis to set the name and all. So is it possible to set date wise logs and test it in development too?

**Reply by Julien-R44** [reply]
Why would you want to send the logs to a file in dev? And if ever you want to, just do the same thing

not sure what you mean by date based logs? there are already timestamps in the logs produced by pino/adonisjs logger. So they'll be in your file too
you can also use this if you want rotation and avoid having too big logs: https://github.com/ExpediaGroup/pino-rotating-file
